timesketch/lib/analyzers/contrib/hashlookup_analyzer.py

```diff
@@ -0,0 +1,115 @@
+"""Index analyzer plugin for Hashlookup."""
+
+import logging
+
+from flask import current_app
+import requests
+
+from timesketch.lib.analyzers import interface
+from timesketch.lib.analyzers import manager
+from timesketch.lib import emojis
+
+
+logger = logging.getLogger("timesketch.analyzers.hashlookup")
+
+
+class HashlookupAnalyzer(interface.BaseAnalyzer):
+    """Analyzer for Hashlookup."""
+
+    NAME = "hashlookup_analyzer"
+    DISPLAY_NAME = "Hashlookup"
+    DESCRIPTION = "Mark events using Hashlookup"
+
+    def __init__(self, index_name, sketch_id, timeline_id=None, **kwargs):
+        """Initialize the Analyzer.
+
+        Args:
+            index_name: OpenSearch index name
+            sketch_id: The ID of the sketch.
+            timeline_id: The ID of the timeline.
+        """
+        super().__init__(index_name, sketch_id, timeline_id=timeline_id)
+        self.hashlookup_url = kwargs.get("hashlookup_url")
+        self.cp = 0
+        self.request_list = list()
+
+    @staticmethod
+    def get_kwargs():
+        """Get kwargs for the analyzer.
+
+        Returns:
+            Info to connect to Hashlookup.
+        """
+
+        hashlookup_url = current_app.config.get("HASHLOOKUP_URL")
+
+        if not hashlookup_url:
+            logger.error("Hashlookup conf not found")
+            return []
+
+        matcher_kwargs = [{"hashlookup_url": hashlookup_url}]
+        return matcher_kwargs
+
+    def get_hash_info(self, hash_value):
+        """Search event on Hashlookup.
+
+        Returns:
+            JSON of Hashlookup's results
+        """
+        results = requests.get(f"{self.hashlookup_url}sha256/{hash_value}")
+
+        if results.status_code != 200:
+            return []
+        if "message" in list(results.json().keys()):
```

Comment:
```suggestion
        if "message" in results.json():
```

AFAIU, this will look for `"message"` in the hashmap instead of a list of those keys, making the lookup much faster.

Reply:
FIxed in 71f1bffe00ecc857c5ad1e9af69818badb06491c